.github/scripts/check-release.sh

```diff
@@ -1,24 +1,41 @@
-#!/bin/bash
-
-# check_tag $current_tag $file_tag $file_name
-function check_tag {
-  if [[ "$1" != "$2" ]]; then
-      echo "Error: the current tag does not match the version in Cargo.toml: found $2 - expected $1"
-      ret=1
-  fi
+#!/usr/bin/env bash
+set -eu -o pipefail
+
+check_tag() {
+    local expected=$1
+    local actual=$2
+    local filename=$3
+
+    if [[ $actual != $expected ]]; then
+        echo >&2 "Error: the current tag does not match the version in $filename: found $actual, expected $expected"
+        return 1
+    fi
+}
+
+read_version() {
+    grep '^version = ' | cut -d \" -f 2
 }
 
+if [[ -z "${GITHUB_REF:-}" ]]; then
+    echo >&2 "Error: GITHUB_REF is not set"
+    exit 1
+fi
+
+if [[ ! "$GITHUB_REF" =~ ^refs/tags/v[0-9]+\.[0-9]+\.[0-9]+(-[a-z0-9]+)?$ ]]; then
+    echo >&2 "Error: GITHUB_REF is not a valid tag: $GITHUB_REF"
+    exit 1
+fi
+
+current_tag=${GITHUB_REF#refs/tags/v}
 ret=0
-current_tag=${GITHUB_REF#'refs/tags/v'}
 
-file_tag="$(grep '^version = ' Cargo.toml | cut -d '=' -f 2 | tr -d '"' | tr -d ' ')"
-check_tag $current_tag $file_tag
+toml_tag="$(cat Cargo.toml | read_version)"
+check_tag "$current_tag" "$toml_tag" Cargo.toml || ret=1
```

Comment:
Technically, the quotes are unnecessary. We've validated `GITHUB_REF`, so we can be sure that `$current_tag` is not an empty string. `$toml_tag` is not empty also.

\<gold-plating\>
And yet, it doesn't hurt to add the double quotes. If `$current_tag` was quoted in the first version of the patch, the error returned by the script would be slightly less opaque.
\</gold-plating\>